spec.html

```diff
@@ -10395,18 +10395,18 @@ <h1>Static Semantics: UTF16Encoding ( _cp_ )</h1>
       <p>The abstract operation UTF16Encoding takes argument _cp_ (a numeric code point value). It performs the following steps when called:</p>
       <emu-alg>
         1. Assert: 0 &le; _cp_ &le; 0x10FFFF.
-        1. If _cp_ &le; 0xFFFF, return _cp_.
+        1. If _cp_ &le; 0xFFFF, return the String value consisting of _cp_.
         1. Let _cu1_ be floor((_cp_ - 0x10000) / 0x400) + 0xD800.
         1. Let _cu2_ be ((_cp_ - 0x10000) modulo 0x400) + 0xDC00.
-        1. Return the code unit sequence consisting of _cu1_ followed by _cu2_.
+        1. Return the string-concatenation of _cu1_ and _cu2_.
```

Comment:
_cu1_ and _cu2_ are code units, and I'm not sure we should directly "concatenate" them. We should instead say

> Return the String value consisting of _cu1_ followed by _cu2_.

or something.

Reply:
The definition of `string-concatenation` specifically allows that an argument can be a code unit, so the string-concatenation of two code units should be fine. (In fact, `the String consisting of the code unit _first_ followed by the code unit _second_` was one of the specific forms that was replaced with an invocation of string-concatenation when it was introduced.)

What has bugged *me* about this operation is that `_cu1_` and `_cu2_` are set to the results of arithmetic formulae, which makes them either Number values or mathematical values (depending on how PR #2007 works out), and usually the spec draws a distinction between such values and code units. (E.g., using phrasing such as "the code unit whose value is [numeric value]" or "the numeric value of [code unit]".) So I'd be inclined to say, e.g.
```
1. Let _cu1_ be the code unit whose value is floor(...) + 0xD800.
```
That's actually independent of this PR, but we got close enough to it that I thought it was worth mentioning.